### PR TITLE
fixes early stopping bug

### PR DESCRIPTION
**Pull request by williamFalcon**

### CODE REVIEW

File: pytorch_lightning/callbacks/early_stopping.py [hunk 1]
```diff
@@ -110,7 +110,8 @@ def on_train_start(self, trainer, pl_module):
         self.best = torch_inf if self.monitor_op == torch.lt else -torch_inf
 
     def on_validation_end(self, trainer, pl_module):
-        self._run_early_stopping_check(trainer, pl_module)
+        import pdb; pdb.set_trace()
```
Comment (Borda) [inline pytorch_lightning/callbacks/early_stopping.py:113]:
```suggestion
```

File: pytorch_lightning/callbacks/early_stopping.py [hunk 2]
```diff
@@ -132,6 +133,7 @@ def _run_early_stopping_check(self, trainer, pl_module):
                 stop_training = True
                 self.on_train_end(trainer, pl_module)
 
+        print(self.wait, self.patience)
```
Comment (Borda) [inline pytorch_lightning/callbacks/early_stopping.py:136]:
```suggestion
```

File: pytorch_lightning/trainer/callback_hook.py [hunk 3]
```diff
@@ -87,6 +87,7 @@ def on_validation_start(self):
             callback.on_validation_start(self, self.get_model())
 
     def on_validation_end(self):
+        import pdb; pdb.set_trace()
```
Comment (Borda) [inline pytorch_lightning/trainer/callback_hook.py:90]:
```suggestion
```

File: pytorch_lightning/trainer/trainer.py [hunk 4]
```diff
@@ -826,6 +826,8 @@ def fit(
         model.logger = self.logger
         self.copy_trainer_model_properties(model)
 
+        import pdb; pdb.set_trace()
+
```
Comment (Borda) [inline pytorch_lightning/trainer/trainer.py]:
```suggestion
```

**Comment by williamFalcon** [comment]
@Borda ... this was WIP... it's done now

**Comment by Borda** [comment]
> @Borda ... this was WIP... it's done now

heh sorry, when y started reading it was not wip anymore :]

**Comment by williamFalcon** [comment]
all good!
Added a test to make sure this doesn't break again lol

**Comment by williamFalcon** [comment]
FYI @jeremyjordan @awaelchli @tullie 
Looks like the fix was very simple lol...

But the problem is that early stopping is not just a simple callback, for instance, it can be disabled temporarily if you set min_epochs, it needs to terminate the program, etc...

We'll have to think harder about how to add to callback system

**Pull request merged**